### PR TITLE
Dark theme and font scaling implementation

### PR DESCRIPTION
**Pull request by nirvn**
This PR implements a dark theme mode for QField as well as font scaling. Both options are available in the QField settings panel:
Screenshot_20230226-163040

Video worth a thousand words:

https://user-images.githubusercontent.com/1728657/221346091-8c4bd5f7-d7dc-4e3a-84ce-3657bd1268d8.mp4

The implementation of a dark theme mode also led to a bunch of tiny UI fixes left and right, it served as a nice cleanup exercise.

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 31ea378f4f3c0fe9e12b74f997796bf9fa69579e: [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3971-f63fc3e39f4e946302938ef88f6fd0751115e30d-arm64-android.apk)
<details>
<summary>Other architectures</summary>

- [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3971-f63fc3e39f4e946302938ef88f6fd0751115e30d-arm-android.apk)
- [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3971-f63fc3e39f4e946302938ef88f6fd0751115e30d-x64-android.apk)
- [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3971-f63fc3e39f4e946302938ef88f6fd0751115e30d-x86-android.apk)
- [**all access arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-3971-f63fc3e39f4e946302938ef88f6fd0751115e30d-arm64-android.apk)
- [all access arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-3971-f63fc3e39f4e946302938ef88f6fd0751115e30d-arm-android.apk)
- [all access x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-3971-f63fc3e39f4e946302938ef88f6fd0751115e30d-x64-android.apk)
- [all access x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-3971-f63fc3e39f4e946302938ef88f6fd0751115e30d-x86-android.apk)
</details>

**Comment by m-kuhn** [comment]
I think there's just one single person able to review this. @marioba can you ?

**Comment by nirvn** [comment]
@suricactus , cloud sync panel now dark theme compatible, thanks for testing.

File: src/qml/DashBoard.qml [hunk 1]
```diff
@@ -319,18 +320,25 @@ Drawer {
               id: temporalButton
               Layout.alignment: Qt.AlignVCenter
               visible: flatLayerTree.isTemporal
-              iconSource: mapSettings.isTemporal ? Theme.getThemeVectorIcon( 'ic_temporal_green_24dp' ) : Theme.getThemeVectorIcon( 'ic_temporal_black_24dp' )
-              bgcolor: "white"
+              iconSource: Theme.getThemeVectorIcon( 'ic_temporal_black_24dp' )
+              iconColor: mapSettings.isTemporal ? Theme.mainColor : Theme.mainTextColor
```
Comment (suricactus) [inline src/qml/DashBoard.qml:324]: what is `Theme.mainColor`, color of what? At least for me, it is confusing especially with `Theme.mainTextColor` as alternative in the condition. Probably `Theme.accentColor`, so at least the "main" word is different?

File: src/qml/LocatorItem.qml [hunk 2]
```diff
@@ -159,81 +160,59 @@ Item {
...
-      height: 24
-      source: Theme.getThemeVectorIcon("ic_qrcode_black_24dp")
-      sourceSize.width: 24 * screen.devicePixelRatio
-      sourceSize.height: 24 * screen.devicePixelRatio
-      fillMode: Image.PreserveAspectFit
-      anchors.centerIn: barcodeReaderButtonRect
-      opacity: 1
-    }
+    iconSource: Theme.getThemeVectorIcon("ic_qrcode_black_24dp")
+    iconColor: Theme.darkTheme ? Theme.mainTextColor : "transparent"
```
Comment (suricactus) [inline src/qml/LocatorItem.qml:208]: hm, so we are hiding this in non-dark DarkTheme? A comment why will help.

**Review by suricactus — APPROVED** [review]
Great job! I have played around with the theme, it looked nice.

Extra points for converting all the tool buttons to `QfToolButton`, we needed this cleanup.

One remark that  you probably already considered: when using bigger fonts, I believe the margins/paddings between the elements should also be scaled up or down. Otherwise the UI looks quite squeezed.

File: src/qml/DashBoard.qml (hunk 1 above)
Comment (nirvn) [reply src/qml/DashBoard.qml:324]: Theme.mainColor is our beautiful lime green, the color to rule them all, the Main Color.

File: src/qml/LocatorItem.qml (hunk 2 above)
Comment (nirvn) [reply src/qml/LocatorItem.qml:208]: I'll add a note in QfToolButton, in this context, Qt handles "transparent" as a do-no-colorize keyword, and the original SVG color is used to render the icon. In this case, the original QR code icon had a nice green touch on top of black, which I'm preserving on light theme.

**Comment by nirvn** [comment]
@suricactus , huge thanks for the review and positive feedback. 

On the padding side, I'm currently 50/50. You're definitively right in mentioning that some elements (eg combo boxes) feel like text is being squeezed in, but I'm worried adding more padding will make it even worse for users who would want large font size due to impairment in a relatively small device size where extra padding would mean an even worse UI.

So ATM, I'd like to give that a try as is.

File: src/qml/DashBoard.qml (hunk 1 above)
Comment (suricactus) [reply src/qml/DashBoard.qml:324]: I guess if I was more involved in the codebase I would have remembered the nickname of the Color of the Colors.

**Pull request merged**